artiq/frontend/artiq_coremgmt.py

```diff
@@ -132,9 +147,29 @@ def main():
             for key, filename in args.file:
                 with open(filename, "rb") as fi:
                     mgmt.config_write(key, fi.read())
+        if args.action == "write_ch_name":
+            ddb = DeviceDB(args.device_db).get_device_db()
+            channel_ntn = channel_number_to_name(ddb)
+            if len(channel_ntn) == 0:
+                print("No device with channel number is found in device_db.py")
+            else:
+                print("Write:")
+                for ch_num, ch_name in channel_ntn.items():
+                    print(f"channel {ch_num}: {ch_name}")
+                    mgmt.config_write("channel "+str(ch_num), ch_name.encode("utf-8"))
         if args.action == "remove":
             for key in args.key:
                 mgmt.config_remove(key)
+        if args.action == "remove_ch_name":
+            ddb = DeviceDB(args.device_db).get_device_db()
```

Comment:
This implementation seems very user-hostile and therefore pretty useless. A common use case to remove and reprogram the channels is when the device database has changed. How would that be done with your proposed implementation?

Reply:
Indeed, my original thought is to remove data before changing device database.
My worry is that some channel data become useless if less channel is found in new device database.
Maybe I add one more entry to store the channel numbers.

Reply:
Or just put the entire channel map into one single config entry.